plugins/modules/web_application_info.yml

```diff
@@ -0,0 +1,103 @@
+---
+# Copyright: (c) 2024, Ansible Project
+# GNU General Public License v3.0+ (see COPYING or https://www.gnu.org/licenses/gpl-3.0.txt)
+
+DOCUMENTATION:
+  module: web_application_info
+  short_description: Get information on IIS Web Applications
+  description:
+    - Returns information about IIS web applications.
+  requirements:
+    - C(IISAdministration) PowerShell module
+  options:
+    name:
+      description:
+        - Name of the web application.
```

Comment:
We should add an extra note here and for `site` explaining what happens if either are unset.